apm-agent-plugins/apm-logging-plugin/apm-logging-plugin-common/src/main/java/co/elastic/apm/agent/loginstr/error/AbstractLoggerErrorCapturingInstrumentation.java

```diff
@@ -89,9 +88,4 @@ public static void logExit(@Advice.Enter @Nullable Object errorCaptureObj) {
         ret.add("logging");
         return ret;
     }
-
-    @Override
-    public ElementMatcher.Junction<ClassLoader> getClassLoaderMatcher() {
-        return not(ofType(nameStartsWith("co.elastic.apm.")));
-    }
```

Comment:
It was required before and now as well for log4j2, because we load it within the agent. However, while it didn't interfere with tests of other frameworks before, it does now, so I removed it from the abstract and implemented it specifically for log4j2, where we test with `TestClassWithDependencyRunner`, which uses `co.elastic.test.ChildFirstURLClassLoader` especially to escape these matchers.